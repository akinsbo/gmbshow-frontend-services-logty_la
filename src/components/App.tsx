import * as React from "react"
import * as styles from "./App.css"
import astImage from "./../images/astronaut.png"
import Data from "./../data.xml"
import { Link } from "react-router-dom"

export interface AppProps {
  compiler: string;
  framework: string;
}

// 'AppProps' describes the shape of props.
// State is never set so we use the '{}' type.
export class App extends React.Component<AppProps, {}> {
  render() {
    console.log(Data)

    return (
      <div className={styles.hello}>
        <nav>
          <Link to="/dashboard">Dashboard</Link>
        </nav>
        <h1>
          Hello from {this.props.compiler} and {this.props.framework}!
        </h1>
        <h1>My Custom React app</h1>
        <img src={astImage} alt="astImage" />
      </div>
    )
  }
}
