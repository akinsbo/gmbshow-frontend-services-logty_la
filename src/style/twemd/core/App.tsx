import * as React from "react"
import { BrowserRouter as Router, Link, Route } from "react-router-dom";
import HomePage from "../pages/HomePage"
import SecondPage from "../pages/SecondPage"
import ThirdPage from "../pages/ThirdPage"
import { APP_DATA } from "../../../../env"
import HtmlHeadModifier from "./HtmlHeadModifier"
import { createUseStyles } from 'react-jss'


const useStyles = createUseStyles({
  hello: {
    color: 'red'
  }
})

// State is never set so we use the '{}' type.
// tslint:disable-next-line:variable-name
const AppRouter = () => {
  const { title, author, link, theme_color } = APP_DATA.siteMetadata
  const classes = useStyles()

  return (
    // Links would be in another file
    <div className={classes.hello}>
      {/* set default header parameters. Pages may individually modify this */}
      <HtmlHeadModifier title={title} author={author} link={link} theme_color={theme_color} />
      <Router basename="/calendar" />
      <Router>
        <nav>
          <Link to="/">HomePage</Link>
          <Link to="/page2/">SecondPage</Link>
          <Link to="/page3/">ThirdPage</Link>
        </nav>

        <Route path="/" exact={true} component={HomePage} />
        <Route path="/page2/" component={SecondPage} />
        <Route path="/page3/" component={ThirdPage} />

      </Router>
      <div>
        <h1>My Custom React app</h1>
      </div>
    </div>
  )
}

export default AppRouter