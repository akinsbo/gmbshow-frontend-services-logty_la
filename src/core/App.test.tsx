import '@testing-library/jest-dom'
import * as React from "react"
import { HelmetProvider } from "react-helmet-async"
import { BrowserRouter as Router } from "react-router-dom"
import * as renderer from "react-test-renderer"
import App from "./App"

describe("App", () => {
  it("should render", () => {
    const renderWrapper = renderer.create(
      <HelmetProvider>
        <Router>
          <App />
        </Router>
      </HelmetProvider>
    )
    expect(renderWrapper).toMatchSnapshot()
  })
})
