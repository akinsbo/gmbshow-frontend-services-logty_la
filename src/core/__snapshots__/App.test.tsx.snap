// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
<div>
  <div>
    <div />
    <nav>
      <a
        href="/"
      >
        HomePage
      </a>
      <a
        href="/page2/"
      >
        SecondPage
      </a>
      <a
        href="/page3/"
      >
        ThirdPage
      </a>
    </nav>
    <div
      class="makeStyles-grow-2"
    >
      <div
        class="header makeStyles-grow-6"
        data-cy="header"
      >
        <header
          class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
          >
            <button
              aria-label="Open drawer"
              class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-7 MuiIconButton-colorInherit MuiIconButton-edgeStart"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <h6
              class="MuiTypography-root makeStyles-titleStyle-9 MuiTypography-h6 MuiTypography-colorInherit MuiTypography-noWrap"
            >
              title
            </h6>
            <div
              class="makeStyles-grow-6"
            />
            <div
              class="makeStyles-sectionDesktop-11"
            />
            <div
              class="makeStyles-sectionMobile-12"
            />
          </div>
        </header>
      </div>
      Loading Base ...
      <div />
      <h1>
        This is the Home page
      </h1>
      <img
        alt="duyonoStonePng"
        src="test-file-stub"
      />
    </div>
    <div>
      <h1>
        My Custom React app
      </h1>
    </div>
  </div>
</div>
`;
